Calibration/IsolatedParticles/plugins/HcalRaddamMuon.cc

```diff
@@ -0,0 +1,807 @@
+  int maxDepth_,type; 
+  edm::Service<TFileService> fs;
+  //////////////////////////////////////////////////////
+  HLTConfigProvider hltConfig_;
+  int               verbosity_;
+  bool              isAOD_, isSLHC_;
+  std::string       hltlabel_;
+  std::vector<std::string> all_triggers,all_triggers1,all_triggers2,all_triggers3,all_triggers4,all_triggers5;
+  ////////////////////////////////////////////////////////////
+
+  std::vector<bool>  isHB, isHE;
+  TTree *TREE;
+  std::vector<bool>  all_ifTriggerpassed;
+  std::vector<bool>  muon_is_good, muon_global, muon_tracker, Trk_match_MuStat;
+  std::vector<int>   hltresults;
+  std::vector<float> energy_hb,time_hb;
+  std::vector<std::string> hltpaths,TrigName_;
+  std::vector<int>    v_RH_h3x3_ieta;
+  std::vector<int>    v_RH_h3x3_iphi;
+  std::vector<double> v_RH_h3x3_ene, PxGlob, PyGlob,PzGlob,Pthetha;
+  std::vector<double>  PCharge,PChi2,PD0, PD0Error,dxyWithBS,dzWithBS,PdxyTrack, PdzTrack,PNormalizedChi2, PNDoF, PValidHits, PLostHits, NPvx, NPvy, NPvz, NQOverP, NQOverPError, NTrkMomentum, NRefPointX, NRefPointY, NRefPointZ;
+  std::vector<bool> NTrkQuality;
+  double h3x3, h3x3Calo; 
+  unsigned int RunNumber, EventNumber , LumiNumber, BXNumber;
+  double _RecoMuon1TrackIsoSumPtMaxCutValue_03, _RecoMuon1TrackIsoSumPtMaxCutValue_04; 
+  int           ntriggers;
+  edm::InputTag HLTriggerResults_;
+  std::string theTrackQuality;
+  edm::InputTag muonsrc_;
+  std::vector <double> track_cosmic_xposition , track_cosmic_yposition, track_cosmic_zposition, track_cosmic_xmomentum,track_cosmic_ymomentum, track_cosmic_zmomentum, track_cosmic_rad, track_cosmic_detid;
+ };
+
+HcalRaddamMuon::HcalRaddamMuon(const edm::ParameterSet& iConfig) {
+  //now do what ever initialization is needed
+  HLTriggerResults_ = iConfig.getUntrackedParameter<edm::InputTag>("HLTriggerResults_");
+  verbosity_        = iConfig.getUntrackedParameter<int>("Verbosity",0);
+  isAOD_            = iConfig.getUntrackedParameter<bool>("IsAOD",false);
+  isSLHC_           = iConfig.getUntrackedParameter<bool>("IsSLHC",true);
+  maxDepth_         = iConfig.getUntrackedParameter<int>("MaxDepth",4);
+  //  muonsrc_           = iConfig.getUntrackedParameter<edm::InputTag>("muonsrc");
+  if (maxDepth_ > 7)      maxDepth_ = 7;
+  else if (maxDepth_ < 1) maxDepth_ = 4;
+}
+
+HcalRaddamMuon::~HcalRaddamMuon() {
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void HcalRaddamMuon::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  
+  clearVectors();
+  RunNumber   = iEvent.id().run();
+  EventNumber = iEvent.id().event();
+  LumiNumber  = iEvent.id().luminosityBlock();
+  BXNumber = iEvent.bunchCrossing();
+ 
+  edm::Handle<edm::PCaloHitContainer> calosimhits;
+  iEvent.getByLabel("g4SimHits","HcalHits",calosimhits);
+ 
```

Comment:
Hi David

I shall add this. I am preparing the next backporting based on 8943. I shall add it there. Would it be OK? Otherwise it will be a huge amount. Regards

Sunanda
________________________________
From: David Lange [notifications@github.com]
Sent: 12 May 2015 14:53
To: cms-sw/cmssw
Cc: Sunanda Banerjee
Subject: Re: [cmssw] bsunanda:Run2-alca9 Backport PRs from 7_5 to handle calibration trigger and AlCa Reco jobs for HCAL (#8752)


In Calibration/IsolatedParticles/plugins/HcalRaddamMuon.cc<https://github.com/cms-sw/cmssw/pull/8752#discussion_r30129923>:

> +//
> +// member functions
> +//
> +
> +// ------------ method called for each event  ------------
> +void HcalRaddamMuon::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
> +
> +  clearVectors();
> +  RunNumber   = iEvent.id().run();
> +  EventNumber = iEvent.id().event();
> +  LumiNumber  = iEvent.id().luminosityBlock();
> +  BXNumber = iEvent.bunchCrossing();
> +
> +  edm::Handle<edm::PCaloHitContainer> calosimhits;
> +  iEvent.getByLabel("g4SimHits","HcalHits",calosimhits);
> +


@bsunanda<https://github.com/bsunanda> - could you add the consumes migration to this module. Thanks

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/8752/files#r30129923>.